### PR TITLE
Add test suite

### PR DESCRIPTION
**Pull request by nwithan8**
Let's make sure that the VCR solution is working properly by adding a comprehensive CI suite to test `EasyVCR` features.

This PR includes:
- Client tests, checking recording/replaying, auto mode, matching rules, and censoring
- VCR tests, checking mode switching, cassette switching, and advanced settings

### CODE REVIEW

**Review by jchen293 — APPROVED** [review]

File: EasyPost.EasyVCR.Tests/ClientTest.cs [hunk 1]
```diff
@@ -6,55 +8,147 @@ namespace EasyPost.EasyVCR.Tests
...
 
-            var fakeDataService = new FakeDataService(client);
+        [TestMethod]
+        public async Task TestAutoMode()
+        {
+            var cassette = TestUtils.GetCassette("test_auto_mode");
+            cassette.Erase(); // Erase cassette before recording
+            Assert.IsTrue(cassette.NumInteractions == 0); // Make sure cassette is empty
+
+            // in replay mode, if cassette is empty, should throw an exception
```
Comment (Justintime50) [inline EasyPost.EasyVCR.Tests/ClientTest.cs:69]: Why are we testing two modes in a single function meant for one mode?

**Review by Justintime50 — COMMENTED** [review]
Looove the comments in tests explaining **why** things are happening!

Only feedback is it appears we are overloading the tests a bit. Can we break them down into smaller testable units? It appears we are sometimes testing two unrelated things per test when a test should only ever be testing one piece of functionality.

File: EasyPost.EasyVCR.Tests/ClientTest.cs (hunk 1 above)
Comment (nwithan8) [reply EasyPost.EasyVCR.Tests/ClientTest.cs:69]: The replay mode test is just a baseline, to show that auto actually does perform differently than replay (auto attempts replay first (which we see results in failure) before recording)

Comment (Justintime50) [reply EasyPost.EasyVCR.Tests/ClientTest.cs:69]: Ok

**Review by Justintime50 — APPROVED** [review]

**Pull request merged**